lib/hammer_cli_foreman_tasks/helper.rb

```diff
@@ -2,7 +2,14 @@ module HammerCLIForemanTasks
   module Helper
     # render the progress of the task using polling to the task API
     def task_progress(task_or_id)
-      task_id = task_or_id.is_a?(Hash) ? task_or_id['id'] : task_or_id
+      task_id = case task_or_id
+      when Hash
+        task_or_id['id']
+      when Array
+        Hash[*task_or_id.flatten(1)]['task_id']
```

Comment:
If we assume it will always be a single element array, can't we just `task_or_id.first['task_id']`?

Reply:
I will try this and see how it works out and if so make the change.